app/actions/trainer.js

```diff
@@ -455,16 +455,22 @@ function handlePogobufError(error) {
   }
 }
 
-function transferPokemon(pokemon, delay) {
+function transferPokemon(selectedPokemon) {
   return async (dispatch) => {
+    const idArray = []
+    const monArray = []
+    selectedPokemon.forEach((currentPokemon) => {
```

Comment:
I recommend we change this to use map.

`const ids = selectedPokemon.map(p => p.id)`

Then later instead of monArray, just reuse the selectedPokemon array.